airflow/models/dagrun.py

```diff
@@ -217,6 +217,12 @@ def set_state(self, state: DagRunState):
             if state == State.QUEUED:
                 self.queued_at = timezone.utcnow()
 
+    @provide_session
+    def reset_run_type(self, session: Session = None):
+        self.run_type = DagRunType.SCHEDULED
+        self.run_id = self.run_id.replace(DagRunType.BACKFILL_JOB.value, DagRunType.SCHEDULED.value)
```

Comment:
Not convinced about this. The run_id is supposed to be an identifier and it probably should not change like this.

Reply:
Yeah, run_id is used in the FK of multiple tables. We can't change the run_id.

Reply:
good call, removed the line.

Reply:
cc @dimonchik-suvorov @zbstof